src/compute/src/compute_state.rs

```diff
@@ -752,17 +791,39 @@ impl PendingPeek {
         let mut l_datum_vec = DatumVec::new();
         let mut r_datum_vec = DatumVec::new();
 
-        while cursor.key_valid(&storage)
-            && if self.peek.key.is_some() {
-                // If `seek_key` tried to seek to a non-existent key, then it landed on a different
-                // key, so we should quit already.
-                *self.peek.key.as_ref().unwrap() == *cursor.get_key(&storage).unwrap()
-            } else {
-                true
+        // We have to sort the literal constraints because cursor.seek_key can seek only forward.
+        self.peek
+            .literal_constraints
+            .iter_mut()
+            .for_each(|vec| vec.sort());
+        let mut literals = LiteralConstraintsIterator::new(&self.peek.literal_constraints);
```

Comment:
Option has an iter methhod: https://doc.rust-lang.org/std/option/enum.Option.html#method.iter

Thus, I think you can just replace this special iterator struct with `self.peek.literal_constraints.iter().flat_map(|l| l)`.

Reply:
Ok, yes, the code became shorter. I'm not really sure why I thought that extra iterator struct to be a good idea.